### DISCUSSION TITLE
[README Incomplete] Unable to resolve dependencies and build rocket-chip

### DISCUSSION THREAD

**Discussion by aliabbasi-engr** [chipsalliance/rocket-chip#3543] [Q&A]
Hi,

I've been trying to make the rocket-chip on Ubuntu, but fail to find the matching versions of all the dependencies.

The latest CI build of the `master` branch is failing according to [its badge](https://github.com/chipsalliance/rocket-chip/workflows/Continuous%20Integration/badge.svg?branch=master). So, I went with the latest release (which I hoped would be a stable version): `v1.6` created on Oct 10, 2022. 

Looking at the [README file of v1.6](https://github.com/chipsalliance/rocket-chip/tree/v1.6), it says install `rocket-tools` (but doesn't mention what version/tag/commit).

So, again on instinct, I went with the latest version, which is [2023.05.02](https://github.com/chipsalliance/rocket-tools/tree/2023.05.02).

The problem is that:

* version `2023.05.02` of `rocket-tools` won't build with gcc-13 (please see my post on Stack Overflow: [Build rocket-tools for gcc-13](https://stackoverflow.com/questions/77596560/build-rocket-tools-for-gcc-13))
* and `v1.6` of `rocket-chip` won't `make` when `riscv` is built by `rocket-tools` with gcc-11 (please see my other post: [Install rocket-chip on Ubuntu](https://stackoverflow.com/questions/77596143/install-rocket-chip-on-ubuntu))

`verilator`, `sbt`, `chisel`, `bison` they all had specific version requirements that was not mentioned clearly in the README files and I had to dig them from I don't know where 😁

Is there a specific setup instruction, distro, or tools, that maintainers follow to get `rocket-chip` to `make` successfully? If so, would it be possible to share that? 🙇

If not, I already have two broken scripts! Would appreciate if maintainers could take a look and see what I'm doing wrong:

* [rocket-chip.jammy.sh](https://gist.github.com/aliabbasi-engr/1951114a3067eddea3025cb253f26cbf) for Ubuntu 22.04 Jammy
* [rocket-chip.mantic.sh](https://gist.github.com/aliabbasi-engr/992c0f0278e032359c25d76395892ba3) for Ubuntu 23.10 Mantic

I'm not biased towards Ubuntu (and any specific version of it). If maintainers recommend another distro which has been tested on, happy to switch.

To be honest, I just need to build a SoC and continue my academic research project 😞. Any help on building the rocket-chip would be highly appreciated. 🙏

P.S. If you think the scripts could be amended and put in the repo, more than happy to help update/test them, and maybe the maintainers could put it in the repo.

**Comment by dohertk** [comment]
FWIW I'm having exactly the same set of problems.  Stuck in unknown dependency hell.

Ubuntu 18.04 and 22.04 both fully updated, and both fail to build either the chip or the tools.

I'm going to be working the issue, and will post back if I have any success.

**Comment by michael-etzkorn** [comment]
Have you tried sorting out dependencies using chipyard? What have you tried
within RC itself? I personally don't use the mill build frequently. It'd be
good to know what sort of issues people are running into on their local
machines.

Toolchain wise I suggest either chipyard's flow to build the riscv
toolchain or [OpenTitan's prebuilt download](
https://opentitan.org/book/doc/getting_started/index.html)


dohertk ***@***.***> 于 2024年1月29日周一 下午9:58写道：
